src/components/mortgage-investment-compare/amortization-table.js

```diff
@@ -128,64 +128,113 @@ const hoverTableCells = (year, option1, option2) => {
  *   @property {Object} shorterOption
  *   @property {Object} longerOption
  *   @property {Object} yearlyResultsByOption
+ *   @property {function} onROISwitch
+ *   @property {function} onInflationSwitch
+ *   @property {boolean} includeROI
+ *   @property {boolean} includeInflation
+ *   @property {Object} input
+ *   @property {function} updtateResult
  */
 
-const AmortizationTable = props => {
-  return (
-    <div>
-      <h4>Yearly Breakdown</h4>
-      <Table bordered responsive className='mt-3'>
-        <Thead>
-          <Tr>
-            <Th colSpan='1' />
-            <Th colSpan='4'>{props.shorterOption.term} year</Th>
-            <Th colSpan='4'>{props.longerOption.term} year</Th>
-          </Tr>
-          <Tr>
-            <Th colSpan='1' />
-            <Th>Mortgage Payment</Th>
-            <Th>Investment Payment</Th>
-            <Th>Loan Amount</Th>
-            <Th>Investment Amount</Th>
-            <Th>Mortgage Payment</Th>
-            <Th>Investment Payment</Th>
-            <Th>Loan Amount</Th>
-            <Th>Investment Amount</Th>
-          </Tr>
-        </Thead>
-        <Tbody>
-          {props.yearlyResultsByOption.shorter.map((_r, year) => {
-            const shorter = props.yearlyResultsByOption.shorter[year]
-            const longer = props.yearlyResultsByOption.longer[year]
-
-            return (
-              <Tr
-                key={'row' + year}
-                className={highlightTableCells(
-                  year,
-                  props.option1,
-                  props.option2
-                )}
-                onMouseOver={() =>
-                  hoverTableCells(year, props.option1, props.option2)
-                }
-              >
-                <Td>Year {year + 1}</Td>
-                <Td>{toUSD(shorter.pmt)}</Td>
-                <Td>{toUSD(shorter.investmentPMT)}</Td>
-                <Td>{toUSD(shorter.loanAmt)}</Td>
-                <Td>{toUSD(shorter.investmentAmt)}</Td>
-                <Td>{toUSD(longer.pmt)}</Td>
-                <Td>{toUSD(longer.investmentPMT)}</Td>
-                <Td>{toUSD(longer.loanAmt)}</Td>
-                <Td>{toUSD(longer.investmentAmt)}</Td>
-              </Tr>
-            )
-          })}
-        </Tbody>
-      </Table>
-    </div>
-  )
+class AmortizationTable extends React.Component {
+  componentWillReceiveProps (nextProps) {
+    if (
+      this.props.includeROI !== nextProps.includeROI ||
+      this.props.includeInflation !== nextProps.includeInflation
+    ) {
+      this.props.updtateResult(nextProps.input)
+    }
+  }
```

Comment:
Was this a performance improvement?

Reply:
I didn't run a profiler, but it's likely insignificant